src/webhook.ts

```diff
@@ -1,26 +1,73 @@
 import { serve } from "https://deno.land/std@0.184.0/http/server.ts";
+import { createEventHandler } from "https://esm.sh/@octokit/webhooks@11.0.0";
+import { verify } from "https://esm.sh/@octokit/webhooks-methods@3.0.2";
 import * as backport from "./backport.ts";
 import * as labels from "./labels.ts";
 import * as mergeQueue from "./mergeQueue.ts";
 
+const secret = Deno.env.get("BACKPORTER_GITHUB_SECRET");
+
 if (
   Deno.env.get("BACKPORTER_GITEA_FORK") === undefined ||
-  Deno.env.get("BACKPORTER_GITHUB_TOKEN") === undefined
+  Deno.env.get("BACKPORTER_GITHUB_TOKEN") === undefined ||
+  secret
 ) {
   console.error(
-    "BACKPORTER_GITEA_FORK and BACKPORTER_GITHUB_TOKEN must be set",
+    "BACKPORTER_GITEA_FORK, BACKPORTER_GITHUB_TOKEN and BACKPORTER_GITHUB_SECRET must be set",
   );
 }
 
-serve((req: Request) => {
-  if (req.url.endsWith("/trigger")) {
+const webhook = createEventHandler({});
+
+webhook.on("push", ({ payload }) => {
+  // on push to main, backport (we have to be careful here as we cherry-pick and
+  // only have one local git repo)
+  if (payload.ref === "refs/heads/main") {
     backport.run();
-    labels.run();
-    mergeQueue.run();
-    return Response.json({
-      message:
-        "Triggered backport, label maintenance, and merge queue PRs sync",
-    });
+  }
+
+  // we should take this opportunity to run the label and merge queue maintenance
+  labels.run();
+  mergeQueue.run();
+});
+
+// on pull request labeling events, run the label and merge queue maintenance
+webhook.on(
+  ["pull_request.labeled", "pull_request.unlabeled"],
+  ({ payload }) => {
+    // these events are very common, so we only run the label and merge queue if
+    // the label is relevant
+    if (labels.isRelevantLabel(payload.label.name)) {
+      labels.run();
+      mergeQueue.run();
+    }
+  },
+);
+
+serve(async (req: Request) => {
+  if (req.url.endsWith("/trigger") && req.method === "POST") {
```

Comment:
Does `req.url` contain query string in Deno? If yes, this could fail.

Reply:
You're right, but that's fine because I don't want it to trigger if there's a query string

Reply:
Might want to add a comment for it.

Reply:
Done